### PR TITLE
Building a simple cryptocurrency blockchain in Nodejs

### LINKED ISSUE #1876: Building a simple cryptocurrency Blockchain in Nodejs
Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Proposed title of article
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*

### Introduction paragraph (2-3 paragraphs):
The unprecedented rise of cryptocurrencies, and their underpinning blockchain technology, have taken the world by storm. The blockchain technology is receiving a lot of attention because of its ability to enhance security in trustful environments, enforce decentralization, and make processes efficient.
Blockchain is the technology that powers digital currencies, such as Bitcoin and Ethereum. It is an innovative distributed public ledger technology that maintains a continuously growing list of records, referred to as blocks, which are connected securely using cryptography.

Traditionally, Python has been the recognized programming language for blockchain development. However, with the proliferation of this amazing technology, the development options have also increased — and Node.js has not been left behind.In this tutorial, I’m going to talk about how to build a simple cryptocurrency blockchain in Node.js and help understand the how blockchain works and it;s implementation in code.

### Key takeaways: 
1. what is blockchain and how it works
2. How to create blockchain
3. Understanding Nodejs 

### References: 
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by yitzackRabin**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1876

### CODE REVIEW

**Comment by yitzackRabin** [comment]
This fixes #1876

**Comment by srishilesh** [comment]
Hey @yitzackRabin,

Welcome to EngEd community!

Could you please add an author bio and your photograph under `/authors` directory? Please refer the contributing guide.

Also, include a hero image of dimensions 600x400 here.

Ping me if any questions

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
+## Building Simple cryptocurrency system using Node.js
+
+In this tutorial, we are going to demonstrate how to build a simple cryptocurrency system, called thecoin.
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:3]: Could you please write a more brief intro about what the goals of this article are? the key takeaways? what are you going to demonstrate? And, explain about "thecoin" in detail

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+![Image title](engineering-education/building-a-simple-cryptocurrency-blockchain/trade.jpg)
+
+We are going to implement the concepts of JavaScript classes and Node.js.
+
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:7]: Add a `Table of contents` section here

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+## Prerequisite
+To follow this tutorial successfully, you’ll need to have the following:
+
+- Node.js Installed in your machine
+
+- Code editor for my case I use VSCode
+
+
+### WHAT IS BLOCKCHAIN
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:16]:
```suggestion
### What is Blockchain?
```

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+![ethereum](engineering-education/building-a-simple-cryptocurrency-blockchain/ethereum.jpg)
+**Blockchain** as the name states it's a block of transactions data growing increasingly to create a chain of transaction occurrences. Valid transaction data are logged into the blockchain network following the peer-to-peer rule laid down by participants.
+
+> Let's get into code
+
+> I'll name my app `thecoin`.
+
+> Create the app named `thecoin.js` and open it in your code editor
+
+> In the folder you're doing current development let's install the crypto library we're going to use using the command
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:28]: You can remove these quotes.

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+npm install --save crypto-js
+```
+
+We'll use this library to import modules in our project.
+
+**Begin**
+
+I'll begin by creating a class 'BlockCypto'.
+
+```Node
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:40]:
```suggestion
```js
```

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        this.index = index;
+        this.current_time = current_time;
+        this.info = info;
+        this.nextHash = nextHash;
+        this.hash = this.computeHash();     
+        }
+        computeHash(){
+            return SHA256(this.info + this.nextHash + this.current_time + JSON.stringify(this.info)).toString();
+        }   
+    }
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:53]: Add comments

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+
+- `info`: Keeps record and provide data about completed transactions 
+
+- `nexthash` : It points to the hash of the next block in the chain, used to keep and maintain the integrity of the blockchain.
+
+- `computeHash` We use this method to calculate the hash of the next block based on its properties.
+
+### The Blockchain Theorem
+
+The **blockchain Theorem** is core. This is where all blocks are connected.
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:78]: Explain in detail about the theorem. Add reference links to it

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+- Code editor for my case I use VSCode
+
+
+### WHAT IS BLOCKCHAIN
+![Bitcoin Image](engineering-education/building-a-simple-cryptocurrency-blockchain/bitcoin.jpg)
+Digital currencies such as Bitcoin, Ethereum are powered and adopted with powerful technology **the Blockchain**. It uses Cryptography to securely connect and maintain a list of records growing continuously known as **Blocks**
+![ethereum](engineering-education/building-a-simple-cryptocurrency-blockchain/ethereum.jpg)
+**Blockchain** as the name states it's a block of transactions data growing increasingly to create a chain of transaction occurrences. Valid transaction data are logged into the blockchain network following the peer-to-peer rule laid down by participants.
+
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:21]: Explain the objective of what you're going to build

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+### The Blockchain Theorem
+
+The **blockchain Theorem** is core. This is where all blocks are connected. 
+
+So let's create a class that will maintain this operation.
+
+Here is the code for the class I named 'Blockchain'
+
+```Node
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:84]:
```suggestion
```js
```

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+        }
+        latestBlock(){
+            return this.block1chain[this.block1chain.length - 1];
+        }
+        addNewBlock(newBlock){
+            newBlock.nextHash = this.latestBlock().hash;
+            newBlock.hash = newBlock.computeHash();        
+            this.block1chain.push(newBlock);
+        }
+    }
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:100]: Add comments

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+
+The main characteristic of blockchain is that once a block has been added to the network, it can't be changed without invalidating the entire blockchain integrity.
+
+To perform this, we use digital security or cryptographic hash which ensures securing and validating of blockchain by producing a new hash every time any change is made in the block thus invalidating the blockchain.
+
+We'll loop over the entire blockchain to check whether any hash has been tampered with, taking into account the exception of the first block which is hardcoded.
+
+Besides, this method will verify if the hashes of each two consecutive blocks are pointing to one another. If the integrity of the blockchain has not been compromised, it returns true; otherwise, in case of any anomalies, it returns false.
+ We'll create this method inside inside `Blockchain` class
+```Node
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:127]:
```suggestion
```js
```

File: articles/building-a-simple-cryptocurrency-blockchain/index.md (hunk 1 above)
Comment (srishilesh) [reply articles/building-a-simple-cryptocurrency-blockchain/index.md:3]: Talk about what blockchain is, then talk about what cryptocurrencies are

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+![Image title](engineering-education/building-a-simple-cryptocurrency-blockchain/trade.jpg)
+
+We are going to implement the concepts of JavaScript classes and Node.js.
+
+## Prerequisite
+To follow this tutorial successfully, you’ll need to have the following:
+
+- Node.js Installed in your machine
+
+- Code editor for my case I use VSCode
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:13]: Also, mention about the technical prerequisites like what concepts would enable the reader to understand this topic better

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @yitzackRabin,

Great article! Please have a look at the suggestions.

Let me know when they're done, we can move on to the tech review.

Thank you

**Comment by yitzackRabin** [comment]
Thanks, I'm working on the changes as soon as possible. Keeping you posted
in 20 minutes

File: articles/building-a-simple-cryptocurrency-blockchain/index.md (hunk 1 above)
Comment (yitzackRabin) [reply articles/building-a-simple-cryptocurrency-blockchain/index.md:3]: Done

Comment (yitzackRabin) [reply articles/building-a-simple-cryptocurrency-blockchain/index.md:3]: Done

**Comment by yitzackRabin** [comment]
fixed #1876

**Comment by yitzackRabin** [comment]
Hello. @srishilesh  I'm done with changes requsted

**Comment by yitzackRabin** [comment]
Hi @srishilesh Is there any other change required?

**Comment by srishilesh** [comment]
> Hi @srishilesh Is there any other change required?
> 

Hey @yitzackRabin,

I'll do the next round of review by tomorrow, and let you know if any changes are needed.

**Comment by yitzackRabin** [comment]
Hi @srishilesh Okay, I'll be waiting

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 13]
```diff
@@ -0,0 +1,187 @@
+## Building Simple cryptocurrency system using Node.js
+
+In this tutorial, we are going to demonstrate how to build a simple cryptocurrency system, called thecoin.
+It is a simple implementation of the of decentralized digital money, based on blockchain technology.
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:4]: The introduction could have been better. Have a look at the intros of other Section articles, to get an idea.

You could introduce by stating about cryptocurrencies, then give 1 - 2 lines explanation about what blockchain is, and then talk about "thecoin".

The introduction and the conclusion are the most important parts of a tutorial. So, you must make sure that the reader must gain interest by reading them.

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 14]
```diff
@@ -0,0 +1,187 @@
...
+- Node.js Installed in your machine
+
+- Code editor for my case I use VSCode
+
+
+### What is Blockchain?
+![Bitcoin Image](engineering-education/building-a-simple-cryptocurrency-blockchain/bitcoin.jpg)
+Digital currencies such as Bitcoin, Ethereum are powered and adopted with powerful technology **the Blockchain**. It uses Cryptography to securely connect and maintain a list of records growing continuously known as **Blocks**
+![ethereum](engineering-education/building-a-simple-cryptocurrency-blockchain/ethereum.jpg)
+**Blockchain** as the name states it's a block of transactions data growing increasingly to create a chain of transaction occurrences. Valid transaction data are logged into the blockchain network following the peer-to-peer rule laid down by participants.
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:30]: Could you explain more briefly? This paragraph can be understood by someone, who already knows what a blockchain is. But, you must assume the perspective of the reader who is beginner.

As a suggestion,
- You could relate blockchain with something else, and give an analogy
- You could talk about its benefits
- Talk about decentralization
- Talk about the use of blockchain in the present world
- Describe about the security system and its peer-peer network.
- Explain its various steps of working.

In fact, you could have separate sections for each of the above topics.

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+
+We'll have good understanding of blockchain and cryptocurrency and it's operation. 
+
+> Let's get into code
+
+> I'll name my app thecoin.
+
+> Create the app named thecoin.js and open it in your code editor
+
+> In the folder you're doing current development let's install the crypto library we're going to use using the command
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:40]:
```suggestion
Let's get into code

I'll name my app `thecoin`.

Create the app named `thecoin.js` and open it in your code editor.

In the development folder, let's install the `crypto` library that we're going to use, using the command:
```

File: articles/building-a-simple-cryptocurrency-blockchain/index.md (hunk 7 above)
Comment (srishilesh) [reply articles/building-a-simple-cryptocurrency-blockchain/index.md:90]: @yitzackRabin  Please have a look at this suggestion. You have not fixed it

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 16]
```diff
@@ -0,0 +1,187 @@
...
+
+- `index`:  This is a distinctive number tracking the index of every block in the blockchain
+
+- `current_time`: As the name states, it keeps a record of the time when each transaction is completed. 
+
+- `info`: Keeps record and provide data about completed transactions 
+
+- `nexthash` : It points to the hash of the next block in the chain, used to keep and maintain the integrity of the blockchain.
+
+- `computeHash` We use this method to calculate the hash of the next block based on its properties.
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:86]: Why are we doing this? What happens? Explain briefly

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 17]
```diff
@@ -0,0 +1,187 @@
...
+As usual. we have our constructor which instantiates the block1chain but this time we passed to it the `initGenesisBlock()` method which initializes the block in the chain. This property refers to an array of blocks in our case.
+
+- `initGenesisBlock()`: This is the first block created in the peer-to-peer network and has not been linked to any other. To our knowledge of indexing it's at index 0, Notice we created it using the previously created 'BlockCrypto' class and passed all the parameters.
+
+- `latestBlock` : As named, we can easily tell it's finding the latest block in the chain. As explained earlier it helps to ensure the hash of the current block and map it to the hash of the previous block to ensure chain integrity.
+
+- `addNewBlock`: This method I used here to add a new block to the chain. Matching the previous hash block and current hash block in our case is to ensure we are not tampering with the chain.
+
+
+Now that our blockchain is ready to work. We are missing something which is a core blockchain principle, the blockchain _integrity_. Let's see how to verify it and test our app.
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:127]: Why are we doing this? Explain the goal

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 18]
```diff
@@ -0,0 +1,187 @@
...
+
+            if(currentBlock.hash !== currentBlock.computeHash()){
+                return false;
+            }
+            if(currentBlock.nextgHash !== nextBlock.hash)
+                return false;
+            }
+            return true;
+        }
+```js
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:153]: I do not find an explanation for this. So, add comments

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 19]
```diff
@@ -0,0 +1,187 @@
...
+Now we can test our app and see the results:
+
+But before we dive into running the code, let's create a new instance of the `Blockchain` class and name it `thecoin`, and add some blocks in the blockchain using random values.  
+
+```Node
+  let thecoin = new Blockchain();
+    thecoin.addNewBlock(new BlockCrypto(1, "06/04/2021", {sender: "Rabin Yitzack", recipient: "Loyd Eve", quantity: 20}));
+    thecoin.addNewBlock(new BlockCrypto(2, "07/04/2021", {sender: "Anita Vyona", recipient: "Felix Mush", quantity: 349}) );
+    console.log(JSON.stringify(thecoin, null, 4));
+```js
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:165]:
```suggestion
```js
  let thecoin = new Blockchain();
    thecoin.addNewBlock(new BlockCrypto(1, "06/04/2021", {sender: "Rabin Yitzack", recipient: "Loyd Eve", quantity: 20}));
    thecoin.addNewBlock(new BlockCrypto(2, "07/04/2021", {sender: "Anita Vyona", recipient: "Felix Mush", quantity: 349}) );
    console.log(JSON.stringify(thecoin, null, 4));
```
```

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 20]
```diff
@@ -0,0 +1,187 @@
...
+
+Typing this command to our terminal `node thecoin.js` will result in this 
+
+![output](engineering-education/building-a-simple-cryptocurrency-blockchain/output.png)
+
+**NOTE:** >Before running the command ensure to navigate on the right path on your terminal.
+
+_hint:_ Using the command `pwd` to check the path.
+
+Here is our full source code
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:177]: You have not added a link to your repository

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 21]
```diff
@@ -0,0 +1,187 @@
...
+
+## Conclusion
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:179]:
```suggestion
### Conclusion
```

File: articles/building-a-simple-cryptocurrency-blockchain/index.md [hunk 22]
```diff
@@ -0,0 +1,187 @@
...
+
+Kudos! You are have just build a simple cryptocurrency app using Node.js, the step is closer to get you started building Pro apps using Node.js, or rather you can just add more features to our simple blockchain and thrill it to the market.
+Nevertheless, I hope that this tutorial has provided you with basic skill proficiency to get you going with the stimulating Node.js development.  
+
+Happy Coding!
+
+
```
Comment (srishilesh) [inline articles/building-a-simple-cryptocurrency-blockchain/index.md:186]: Add a `Further reading` section, that contains references or other links that would help the reader understand the topic better

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @yitzackRabin,

I have suggested a few more major changes. Please have a look at my suggestions, and resolve them, before we proceed to the final review.

I think you have added the `hero` image in the parent directory. Instead, I want you to add it to your tutorial directory.

Have a look at other Section articles to see the standard of writing. It would acts as a guide to improve your article.

Thank you.

File: articles/building-a-simple-cryptocurrency-blockchain/index.md (hunk 16 above)
Comment (yitzackRabin) [reply articles/building-a-simple-cryptocurrency-blockchain/index.md:86]: > Why are we doing this? What happens? Explain briefly

Hi @srishilesh I've just made explanation of the code below, Are you able to understand it?

File: articles/building-a-simple-cryptocurrency-blockchain/index.md (hunk 17 above)
Comment (yitzackRabin) [reply articles/building-a-simple-cryptocurrency-blockchain/index.md:127]: > Why are we doing this? Explain the goal
@srishilesh I'm trying to explain the code below there.

File: articles/building-a-simple-cryptocurrency-blockchain/index.md (hunk 18 above)
Comment (yitzackRabin) [reply articles/building-a-simple-cryptocurrency-blockchain/index.md:153]: Adding comments asap!

File: articles/building-a-simple-cryptocurrency-blockchain/index.md (hunk 16 above)
Comment (srishilesh) [reply articles/building-a-simple-cryptocurrency-blockchain/index.md:86]: Hey,

I am able to understand it. But, the reader might not. That's why I suggested.

**Comment by yitzackRabin** [comment]
Is there any simple language I can use sir, Kindly assist on this if there
is.

**Comment by yitzackRabin** [comment]
hello @srishilesh Kindly check the updates and keep me posted on any change to be done.

File: articles/building-a-simple-cryptocurrency-blockchain/index.md (hunk 14 above)
Comment (yitzackRabin) [reply articles/building-a-simple-cryptocurrency-blockchain/index.md:40]: Hi @srishilesh  Just provided it on the introduction part

**Comment by yitzackRabin** [comment]
Hey,
@srishilesh I've added some description, Kindly check with me and Update. Thanks

**Review by srishilesh — APPROVED** [review]
Hey @yitzackRabin,

Thanks for making the suggested changes.

Great work. I've learnt a lot from your article.

Please go through all my commits carefully, to understand on how you could improve on your future articles.

Moving this to the final review.

A few general suggestions:
- Make use of grammarly for grammar checks.
- Use [Hemingwapp](https://hemingwayapp.com) for writing short and clear sentences.
- Add in detail explanations of code.
- Avoid the usage of "I".
- After you finish writing the article, re-read from the perspective of a beginner reader.
- Make a note of different sections used in this tutorial.

Thank you.

**Comment by yitzackRabin** [comment]
Hey @srishilesh Thanks for your suggestions they are of help to me and I hope i'll improve on the next article

**Comment by hectorkambow** [comment]
👋 Good afternoon - and welcome to the EngEd program.
I'll be running a final review on your PR when ready.
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on [plagiarism ](https://www.scribbr.com/category/plagiarism/)and what is considered as such.

@yitzackRabin 
[Building a simple cryptocurrency blockchain in Nodejs #1991.pdf](https://github.com/section-engineering-education/engineering-education/files/6354545/Building.a.simple.cryptocurrency.blockchain.in.Nodejs.1991.pdf)

**Comment by yitzackRabin** [comment]
hi @Hectorkabow Acknowledged.

**Comment by yitzackRabin** [comment]
Hi @hectorkambow I'm with the chnges requested. Thanks

**Comment by yitzackRabin** [comment]
hi @Hectorkabow all updates made. thanks

**Pull request merged**